daemon/cmd/cni.go

```diff
@@ -121,7 +125,9 @@ func (d *Daemon) startCNIConfWriter(opts *option.DaemonConfig, cleaner *daemonCl
 
 	// We used to disable CNI generation with this environment variable
 	// It's no longer documented, but we need to still support it.
-	if os.Getenv("CILIUM_CUSTOM_CNI_CONF") == "true" {
+	// If CILIUM_CUSTOM_CNI_CONF == true, we want that cni-install.sh will not generate the 05-cilium.conf or 05-cilium.conflist
+	// But we want cilium-agent can install cni according to opts.ReadCNIConfiguration
+	if os.Getenv("CILIUM_CUSTOM_CNI_CONF") == "true" && opts.ReadCNIConfiguration == "" {
```

Comment:
Oh, good catch; this was a mistake; I'll fix it in https://github.com/cilium/cilium/pull/22035